foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/oxm/util/NamespaceResolverStorage.java

```diff
@@ -0,0 +1,159 @@
+package org.eclipse.persistence.internal.oxm.util;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.Vector;
+import java.util.function.BiFunction;
+
+import org.eclipse.persistence.internal.oxm.Namespace;
+
+import static org.eclipse.persistence.internal.oxm.util.VectorUtils.emptyVector;
+import static org.eclipse.persistence.internal.oxm.util.VectorUtils.unmodifiableVector;
+
+public class NamespaceResolverStorage extends LinkedHashMap<String, String> {
+    private static final long serialVersionUID = -4697397620139076774L;
+    private transient Vector namespaces = emptyVector();
+    private transient boolean modified = false;
+
+    public NamespaceResolverStorage() {
+        super();
+    }
+
+    public NamespaceResolverStorage(int initialCapacity) {
+        super(initialCapacity);
+    }
+
+    private void readObject(java.io.ObjectInputStream in)
+        throws IOException, ClassNotFoundException {
+        in.defaultReadObject();
+        namespaces = emptyVector();
+        modified = true;
+    }
```

Comment:
Is this method needed?

Reply:
This is a part of the Serializable interface implementation. Seems in some cases it was used for deserialization. Unfortunately the testing was done long ago, so I'm not sure now.